### PR TITLE
Fix `y_huff`, Rewrite `caravel_user_project_example_full/harden.py`, Remove IPM manifest (redundant)

### PR DESCRIPTION
**Pull request by donn**

### CODE REVIEW

File: y_huff/config.json [hunk 1]
```diff
@@ -8,16 +8,16 @@
     "GPL_CELL_PADDING": 4,
     "DPL_CELL_PADDING": 4,
     "FP_CORE_UTIL": 25,
+    "QUIT_ON_SYNTH_CHECKS": false,
```
Comment (kareefardi) [inline y_huff/config.json:11]: Can't we fix the errors?

File: caravel_user_project_example_full/harden.py [hunk 2]
```diff
@@ -1,106 +1,95 @@
...
+        design_dir=upe_dir,
+        pdk="sky130A",
+        pdk_root=pdk_root,
+    )
+    upe_state_out = upe_flow.start(tag=run_tag)
+
+    user_proj_example = Macro.from_state(upe_state_out)
+
+    # TODO: Fix these hacks
+    user_proj_example.nl = [x.replace(".nl.v", ".pnl.v") for x in user_proj_example.nl]
```
Comment (kareefardi) [inline caravel_user_project_example_full/harden.py:60]: Why is the hack needed?

File: y_huff/config.json (hunk 1 above)
Comment (donn) [reply y_huff/config.json:11]: It's not worth the time investment.

File: caravel_user_project_example_full/harden.py (hunk 2 above)
Comment (donn) [reply caravel_user_project_example_full/harden.py:60]: Long story short, until efabless/openlane2#369 is merged, OpenLane does not store powered netlists in the macro object, however, this causes an issue with the JSON header generation script which sees two pins that do not exist in the non-powered netlist.

File: y_huff/config.json (hunk 1 above)
Comment (kareefardi) [reply y_huff/config.json:11]: We should add a comment describing the errors and why were they waived.

Comment (donn) [reply y_huff/config.json:11]: This PR gates every other PR. Can we get it merged and defer these? There are other designs that would also need this treatment should you opt to do it.

Comment (kareefardi) [reply y_huff/config.json:11]: ok

**Pull request merged**